### PR TITLE
Allowing more than 4 dots 

### PR DESCRIPTION
**Pull request by openGeeksLab**
Why don't you allow more than 4 dots?
for example
 https://www.google.com/search?sxsrf=ACYBGNTJFmX-GjNJ8fM-2LCkqyNyxGU1Ng%3A1575534146332&ei=Qr7oXf7rE4rRrgSEgrmoAw&q=clover&oq=clover&gs_l=psy-ab.3..0i67j0l9.2986.3947..4187...0.2..0.281.1366.1j0j5......0....1..gws-wiz.......0i71j35i39j0i131.qWp1nz4IJVA&ved=0ahUKEwj-lP6Iip7mAhWKqIsKHQRBDjUQ4dUDCAs&uact=5
It's just google links and there are more than 4 dots, is it wrong link for you?

with help just a little change I've given the opportunity to recognize such links

Please merge it, or explain to me why it works in such a way

### CODE REVIEW

**Comment by puzrin** [comment]
Such kind of modules should not produce false positive (fetch tail which is not link). So, it's very dangerous to create too wide rules in anvance.

Approach is to create as narrow rules as possible, and extend those after bug reports if no conflicts found.

I understand your case, and have questions:

- Can those links have dots at the end?
- What is max number of dots?

**Comment by openGeeksLab** [comment]
@puzrin Thank you for a quick answer.
1) The dots in the end, can't have a payload, so they shouldn't be parsed
2) If we're talking about links from google search, it can be any amount of dots, it directly depends on the request 
 for example 
https://www.google.com/search?sxsrf=ACYBGNQIsAdD2HHcLKxVjn2b6o55dScWaA%3A1575558086811&ei=xhvpXbOAMaiorgSExIaACQ&q=interesting+stories.......&oq=interesting+stories.......&gs_l=psy-ab.3..0i22i30l10.2395.12446..15641...7.2..0.121.1331.13j1......0....1..gws-wiz.......0i71j35i39j0i67j0j0i203.CRMR3t9i-rw&ved=0ahUKEwiz-9ig457mAhUolIsKHQSiAZAQ4dUDCAs&uact=5

File: lib/re.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ module.exports = function (opts) {
           '\\"(?:(?!' + re.src_ZCc + '|["]).)+\\"|' +
           "\\'(?:(?!" + re.src_ZCc + "|[']).)+\\'|" +
           "\\'(?=" + re.src_pseudo_letter + '|[-]).|' +  // allow `I'm_king` if no pair found
-          '\\.{2,4}[a-zA-Z0-9%/]|' + // github has ... in commit range links,
+          '\\.{2,}[a-zA-Z0-9%/]|' + // github has ... in commit range links,
```
Comment (puzrin) [inline lib/re.js]: 1. Add note `google has many dots in "google search" links`.
2. Add your example link to tests.

File: lib/re.js [hunk 2]
```diff
@@ -56,8 +56,7 @@ module.exports = function (opts) {
           '\\"(?:(?!' + re.src_ZCc + '|["]).)+\\"|' +
           "\\'(?:(?!" + re.src_ZCc + "|[']).)+\\'|" +
           "\\'(?=" + re.src_pseudo_letter + '|[-]).|' +  // allow `I'm_king` if no pair found
-          '\\.{2,4}[a-zA-Z0-9%/]|' + // github has ... in commit range links,
-                                     // google has .... in links (issue #66)
+          '\\.{2,}[a-zA-Z0-9%/]|' + // google has many dots in "google search" links.
```
Comment (puzrin) [inline lib/re.js]: Please, don't delete existing comments, those are important. Just add one more line.

**Comment by puzrin** [comment]
https://github.com/markdown-it/linkify-it/blob/master/test/fixtures/links.txt#L168 - here is place to insert your link (with one empty line before and after)

**Comment by openGeeksLab** [comment]
I had to change regex a little, tests were passed, please check

**Comment by puzrin** [comment]
> I had to change regex a little, tests were passed, please check

This rules are euristic, very fragile. We can not do "little changes" without explanation. Need details, why additional change was done.

**Comment by openGeeksLab** [comment]
yeah, I agree with you.
I need time to figure out into it more deeply.
What if I remove these last changes, and will prepare new PR for this? 
Will it be ok for you?

**Comment by puzrin** [comment]
> What if I remove these last changes, and will prepare new PR for this?
> Will it be ok for you?

You can just rewrite history and do forced push here, in the same PR. That will work.

As far as i underastand, only 3 changes needed:

1. Remove `4` from regexp
2. Add comment about multiple docs
3. Add your link to tests

If tests passe - i'm ok to merge. If more regexp changes needed - that should be reviewed/discussed.

File: test/fixtures/links.txt [hunk 3]
```diff
@@ -166,6 +166,10 @@ https://github.com/markdown-it/linkify-it/compare/360b13a733f521a8d4903d3a5e1e46
 
 https://www.google.com/search?source=hp&ei=y0QtW7yUGpiq0PEPv5qv8Ag&q=best+sci+fi+books&oq=best+sci&gs_l=psy-ab.3.1.0j0i20i264k1j0j0i131k1j0l6.2035.4079.0.5985.9.8.0.0.0.0.132.797.5j3.8.0....0...1c.1.64.psy-ab..1.8.797.0..35i39k1j0i67k1.0.iug3EHWIyxc
 
+https://www.google.com/search?sxsrf=ACYBGNTJFmX-GjNJ8fM-2LCkqyNyxGU1Ng%3A1575534146332&ei=Qr7oXf7rE4rRrgSEgrmoAw&q=clover&oq=clover&gs_l=psy-ab.3..0i67j0l9.2986.3947..4187...0.2..0.281.1366.1j0j5......0....1..gws-wiz.......0i71j35i39j0i131.qWp1nz4IJVA&ved=0ahUKEwj-lP6Iip7mAhWKqIsKHQRBDjUQ4dUDCAs&uact=5
+
+https://www.google.com/search?sxsrf=ACYBGNQIsAdD2HHcLKxVjn2b6o55dScWaA%3A1575558086811&ei=xhvpXbOAMaiorgSExIaACQ&q=interesting+stories.......&oq=interesting+stories.......&gs_l=psy-ab.3..0i22i30l10.2395.12446..15641...7.2..0.121.1331.13j1......0....1..gws-wiz.......0i71j35i39j0i67j0j0i203.CRMR3t9i-rw&ved=0ahUKEwiz-9ig457mAhUolIsKHQSiAZAQ4dUDCAs&uact=5
+
```
Comment (puzrin) [inline test/fixtures/links.txt]: Are 2 new links really needed? 1 new link not enougth?

Comment (openGeeksLab) [reply test/fixtures/links.txt]: @puzrin I agree with you will do it

**Comment by openGeeksLab** [comment]
@puzrin please, check

**Comment by oleg-rdk** [comment]
@puzrin is there something else that holds this PR? if so, let us know, so we can help & merge it quicker

**Pull request merged**